### PR TITLE
Export options and basic export report dto

### PR DESCRIPTION
**Pull request by inv-jishnu**
## Description

Added export options builder and export report class

## Related issues and/or PRs

PR dependency
- https://github.com/scalar-labs/scalardb/pull/2388

## Changes made

Added export options builder and export report class

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes.
- [ ] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

This PR is part of the on-going process of merging all ScalarDB Data Loader code into the main repository.

## Release notes

NA

### CODE REVIEW

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

File: data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataexport/ExportReport.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * @author Jishnu J
+ */
+public class ExportReport {
+
+  /**
+   * The field is used to get the total number of rows exported from the table and written to the
+   * exported file. AtomicInteger is used because it is thread safe and this parameter is a counter
+   * which is updated inside threads
+   */
+  private final AtomicInteger exportedRowCount;
```
Comment (brfrn169) [inline data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataexport/ExportReport.java:17]: `LongAdder` might be useful here:
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/LongAdder.html

File: data-loader/core/src/test/java/com/scalar/db/dataloader/core/dataexport/ExportReportTest.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.scalar.db.dataloader.core.dataexport;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ExportReportTest {
```
Comment (brfrn169) [inline data-loader/core/src/test/java/com/scalar/db/dataloader/core/dataexport/ExportReportTest.java:6]:
```suggestion
class ExportReportTest {
```

**Review by brfrn169 — COMMENTED** [review]
Left a couple of comments. Please take a look when you have time!

File: data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataexport/ExportReport.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (inv-jishnu) [reply data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataexport/ExportReport.java]: I have moved the initialisation as suggested (I have changed AtomicInteger to LongAdder based on Suzuki-san suggestion).
Thank you.

File: data-loader/core/src/test/java/com/scalar/db/dataloader/core/dataexport/ExportReportTest.java (hunk 2 above)
Comment (inv-jishnu) [reply data-loader/core/src/test/java/com/scalar/db/dataloader/core/dataexport/ExportReportTest.java:6]: I have updated it as suggested.
Thank you.

File: data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataexport/ExportReport.java (hunk 1 above)
Comment (inv-jishnu) [reply data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataexport/ExportReport.java]: Thank you for the suggestion.
I have replaced `AtomicInteger` with `LongAdder`.

File: data-loader/core/src/test/java/com/scalar/db/dataloader/core/dataexport/ExportReportTest.java (hunk 2 above)
Comment (brfrn169) [reply data-loader/core/src/test/java/com/scalar/db/dataloader/core/dataexport/ExportReportTest.java:6]: It looks like it's not updated?

Comment (inv-jishnu) [reply data-loader/core/src/test/java/com/scalar/db/dataloader/core/dataexport/ExportReportTest.java]: @brfrn169 
My apologies, I think I may have missed to commit the file. I have pushed the change now.

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**